app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -52,8 +55,22 @@ def finalize
 
         Rails.logger.debug "cleaning cache for keys that begin with 'job_invocation_#{job_invocation.id}'"
         Rails.cache.delete_matched(/\A#{JobInvocation::CACHE_PREFIX}_#{job_invocation.id}/)
-        # creating the success notification should be the very last thing this tasks do
+      end
+
+      def notify_on_success(plan)
         job_invocation.build_notification.deliver!
+
+        if ['Subscribe to my succeeded jobs', 'Subscribe to all my jobs'].include?(mail_notification_preference&.interval)
+          RexJobMailer.job_finished(job_invocation, subject: _("REX job has succeeded - %s") % job_invocation.to_s).deliver_now
+        end
+      end
+
+      def notify_on_failure(plan)
+        job_invocation.build_notification.deliver!
+
+        if ['Subscribe to my failed jobs', 'Subscribe to all my jobs'].include?(mail_notification_preference&.interval)
+          RexJobMailer.job_finished(job_invocation, subject: _("REX job has failed - %s") % job_invocation.to_s).deliver_now
+        end
       end
```

Comment:
I had in mind something like this, but don't feel too strongly about it

```suggestion
      def notify_on_finish(plan)
        job_invocation.build_notification.deliver!

        event = plan.result == 'success' ? 'succeeded' : 'failed'
        if ["Subscribe to my #{event} jobs", 'Subscribe to all my jobs'].include?(mail_notification_preference&.interval)
          RexJobMailer.job_finished(job_invocation, subject: _("REX job has %{event} - %{job}") % {:job => job_invocation.to_s, :event => event}).deliver_now
        end
    end
```